src/components/molecules/Visualizer/hooks.ts

```diff
@@ -210,31 +238,47 @@ export default ({
 };
 
 function useLayers({
-  layers,
+  rootLayer,
   selected: outerSelectedPrimitiveId,
   onSelect,
 }: {
-  layers: LayerStore;
+  rootLayer: Layer | undefined;
   selected?: string;
   onSelect?: (id?: string, options?: SelectLayerOptions) => void;
 }) {
   const [selectedLayerId, innerSelectLayer] = useState<string | undefined>();
   const [layerSelectionReason, setSelectionReason] = useState<string | undefined>();
   const [layerOverridenInfobox, setPrimitiveOverridenInfobox] = useState<OverriddenInfobox>();
+  const [layers] = useState<LayerStore>(() => new LayerStore(rootLayer));
+  const forceUpdate = useUpdate();
+
+  useMemo(() => {
+    layers.setDatabaseLayers(rootLayer);
+    forceUpdate();
+  }, [layers, rootLayer, forceUpdate]);
 
   const selectedLayer = useMemo(
-    () => (selectedLayerId ? layers.findById(selectedLayerId) : undefined),
+    () => (selectedLayerId ? layers?.findById(selectedLayerId) : undefined),
     [selectedLayerId, layers],
   );
 
   const selectLayer = useCallback(
     (id?: string, { reason, overriddenInfobox }: SelectLayerOptions = {}) => {
       innerSelectLayer(id);
-      onSelect?.(id);
+      id && !!layers.findById(id) && onSelect?.(id);
```

Comment:
Is it okay, for example, if `!!layers.findById(id)` is false for the other methods inside this useCallback to be called without `onSelect`?
Or is it better to move this check to the top of the useCallback like:
`if (!id || !layers.findById(id)) return;` so that its all or nothing?